connectors/sources/gcs.py

```diff
@@ -0,0 +1,311 @@
+
+        Yields:
+            Dictionary: Contains the list of fetched blobs from Google Cloud Storage.
+        """
+        for bucket in buckets.get("items", []):
+            async for blob in self._api_call(
+                resource="objects",
+                method="list",
+                full_response=True,
+                bucket=bucket["id"],
+                userProject=self.user_project_id,
+            ):
+                yield blob
+
+    def prepare_blob_document(self, blob):
+        """Apply key mappings to the blob document.
+
+        Args:
+            blob (dictionary): Blob's metadata returned from the Google Cloud Storage.
+
+        Returns:
+            dictionary: Blobs metadata mapped with the keys of `BLOB_ADAPTER`.
+        """
+        blob_document = {}
+        for elasticsearch_field, google_cloud_storage_field in BLOB_ADAPTER.items():
+            blob_document[elasticsearch_field] = blob.get(google_cloud_storage_field)
+        blob_name = urllib.parse.quote(blob_document["name"], safe="'")
+        blob_document[
+            "url"
+        ] = f"{CLOUD_STORAGE_BASE_URL}{blob_document['bucket_name']}/{blob_name};tab=live_object?project={self.user_project_id}"
+        return blob_document
+
+    def get_blob_document(self, blobs):
+        """Generate blob document.
+
+        Args:
+            blobs (dictionary): Dictionary contains blobs list.
+
+        Yields:
+            dictionary: Blobs metadata mapped with the keys of `BLOB_ADAPTER`.
+        """
+        for blob in blobs.get("items", []):
+            yield self.prepare_blob_document(blob=blob)
+
+    async def get_content(self, blob, timestamp=None, doit=None):
+        """Extracts the content for allowed file types.
+
+        Args:
+            blob (dictionary): Formatted blob document.
+            timestamp (timestamp, optional): Timestamp of blob last modified. Defaults to None.
+            doit (boolean, optional): Boolean value for whether to get content or not. Defaults to None.
+
+        Returns:
+            dictionary: Content document with id, timestamp & text
+        """
+        if not (
+            self.enable_content_extraction
+            and doit
+            and os.path.splitext(blob["name"])[-1] in SUPPORTED_FILETYPE
+            and int(blob["size"])
+        ):
+            return
+
+        if int(blob["size"]) > DEFAULT_FILE_SIZE_LIMIT:
+            logger.warning(
+                f"File size {int(blob['size'])} of file {blob['name']} is larger than {DEFAULT_FILE_SIZE_LIMIT} bytes. Discarding the file content"
+            )
+            return
+
+        content_text = await self._api_call(
+            resource="objects",
+            method="get",
+            bucket=blob["bucket_name"],
+            object=blob["name"],
+            alt="media",
+            userProject=self.user_project_id,
+        ).__anext__()
```

Comment:
you can use `anext()` see https://docs.python.org/3/library/functions.html#anext